sources/src/ChatGPTPopover.tsx

```diff
@@ -1,79 +1,143 @@
 import {
   AppBar,
   AppBarProps,
-  Backdrop,
   IconButton,
   paperClasses,
   Popover,
   PopoverProps,
   Typography
 } from '@mui/material';
 import CloseRounded from '@mui/icons-material/CloseRounded';
+import RemoveRounded from '@mui/icons-material/RemoveRounded';
 import ChatGPT, { ChatGPTProps } from './ChatGPT.tsx';
-import React from 'react';
+import React, { ReactNode, useCallback, useEffect } from 'react';
+import MinimizedBar from '@craftercms/studio-ui/components/MinimizedBar';
+
+interface ChatGPTAppBarProps extends AppBarProps {
+  children: ReactNode;
+}
+
+function ChatGPTAppBar({ children, ...appBarProps }: Readonly<ChatGPTAppBarProps>) {
+  return (
+    <AppBar
+      position="static"
+      color="inherit"
+      {...appBarProps}
+      sx={{
+        display: 'flex',
+        flexDirection: 'row',
+        placeContent: 'space-between',
+        alignItems: 'center',
+        pl: 2,
+        pr: 1,
+        py: 1,
+        ...appBarProps?.sx
+      }}
+    >
+      {children}
+    </AppBar>
+  );
+}
 
 export interface ChatGPTPopoverProps extends PopoverProps {
   appBarTitle?: string;
   width?: number | string;
   height?: number | string;
   appBarProps?: AppBarProps;
-  chatGPTProps: ChatGPTProps;
+  chatGPTProps?: ChatGPTProps;
+  isMinimized?: boolean;
+  onMinimize?: () => void;
+  onMaximize?: () => void;
 }
 
-function ChatGPTPopover(props: ChatGPTPopoverProps) {
+function ChatGPTPopover(props: Readonly<ChatGPTPopoverProps>) {
   const {
     open,
     onClose,
     appBarProps,
     chatGPTProps,
+    isMinimized = false,
+    onMinimize,
+    onMaximize,
     appBarTitle = 'AI Assistant',
     width = 450,
     height = 500,
     ...popoverProps
   } = props;
+
+  const handleMinimize = useCallback(() => {
+    onMinimize?.();
+  }, [onMinimize]);
+
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && !isMinimized) {
+        handleMinimize();
+      }
+    };
+
+    if (open) {
+      document.addEventListener('keydown', handleKeyDown);
+    }
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [open, isMinimized, handleMinimize]);
```

Comment:
@russdanner I think the minimise is good but not too sure about completely changing the behaviours used all around for escape and backdrop click. Perhaps register a minimise keyboard shortcut for minimising but don't feel confident about changing the close behaviour of the escape and backdrop.

Reply:
@russdanner @rart We can disable the behavior to minimize when clicking backdrop or use the escape key. However, this will cause the chat history to be lost if the user clicks on it by mistake.

Reply:
We could also add a close confirmation all around for the close triggers.

```
Confirm closing this chat?
Closing would discard the current conversation.
[Close]
[Minimise]
[Cancel]
```

Reply:
Updated to use `AlertDialog`